### PR TITLE
Auth backend fix

### PR DESCRIPTION
**Pull request by Bomoga**
- [x] Understand the codebase and auth endpoint implementation
- [x] Fix whitespace-only password validation bug in `auth.py` (added `.strip()` to password check)
- [x] Add tests for `POST /api/v1/auth/register` (success, empty username, whitespace username, empty password, whitespace password, duplicate username → 409)
- [x] Add tests for `POST /api/v1/auth/token` (success, wrong password → 401, unknown user → 401)
- [x] All 10 API tests pass

---

📱 Kick off Copilot coding agent tasks wherever you are with [GitHub Mobile](https://gh.io/cca-mobile-docs), available on iOS and Android.

### CODE REVIEW

File: backend/src/api/routers/auth.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Invalid username or password",
+            headers={"WWW-Authenticate": "Bearer"},
+        )
+    payload = {
+        "sub": user["id"],
+        "exp": datetime.datetime.now(datetime.UTC) + datetime.timedelta(hours=24),
+    }
+    token = jwt.encode(payload, settings.JWT_SECRET, algorithm="HS256")
+    return TokenResponse(access_token=token)
```
[the remark this answers is not in the document]
Comment (Bomoga) [reply backend/src/api/routers/auth.py:62]: @copilot apply changes based on [this feedback](https://github.com/Bomoga/NoteBud/pull/123#discussion_r3006722613)
